tests/testthat/test-deployments.R

```diff
@@ -0,0 +1,96 @@
+test_that("deployments() works empty data frame if no deployments", {
+  dir <- local_temp_app()
+  out <- deployments(dir)
+  expect_s3_class(out, "data.frame")
+  expect_equal(nrow(out), 0)
+})
+
+test_that("deployments() can filter", {
+  mockr::local_mock(accounts = fakeAccounts(c("foo", "bar"), c("foo", "bar")))
+  dir <- local_temp_app()
+  saveDeployment(
+    dir,
+    createDeploymentTarget(
+      appName = "my-app",
+      appTitle = "",
+      appId = 10,
+      account = "foo",
+      username = "foo",
+      server = "foo"
+    ),
+    application = list(),
+    hostUrl = NA
+  )
+  saveDeployment(
+    dir,
+    createDeploymentTarget(
+      appName = "my-app2",
+      appTitle = "",
+      appId = 10,
+      account = "bar",
+      username = "bar",
+      server = "bar"
+    ),
+    application = list(id = "123"),
+    hostUrl = NA
+  )
+
+  out <- deployments(dir)
+  expect_s3_class(out, "data.frame")
+  expect_equal(nrow(out), 2)
+
+  out <- deployments(dir, nameFilter = "my-app")
+  expect_equal(nrow(out), 1)
+
+  out <- deployments(dir, serverFilter = "foo")
+  expect_equal(nrow(out), 1)
+
+  out <- deployments(dir, accountFilter = "foo")
+  expect_equal(nrow(out), 1)
```

Comment:
These tests cover the number of records, but not their shape.

These tests don't seem to cover the widening of metadata, proving that all columns are always returned, even when the original dcf contains a subset of fields.

Reply:
That's a bit hard to test at the moment because of the way these functions are structured; let me do a bit more refactoring and I'll include in the next PR